orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/CapabilitiesController.java

```diff
@@ -46,4 +55,30 @@ public CapabilitiesController(
         .map(DeploymentMonitorDefinition::new)
         .collect(Collectors.toList());
   }
+
+  @GetMapping("/capabilities/expressions")
+  public ExpressionCapabilityResult getExpressionCapabilities() {
+    ExpressionCapabilityResult result = new ExpressionCapabilityResult();
+
+    for (ExpressionFunctionProvider provider :
+        expressionsSupport.getExpressionFunctionProviders()) {
+      Collection<ExpressionFunctionProvider.FunctionDefinition> functions =
+          provider.getFunctions().getFunctionsDefinitions();
+
+      for (ExpressionFunctionProvider.FunctionDefinition function : functions) {
+        result
+            .getFunctions()
+            .add(new ExpressionFunctionDefinition(provider.getNamespace(), function));
+      }
+    }
+
+    for (PipelineExpressionEvaluator.SpelEvaluatorVersion spelVersion :
+        PipelineExpressionEvaluator.SpelEvaluatorVersion.values()) {
+      if (spelVersion.getIsSupported()) {
+        result.getSpelEvaluators().add(spelVersion.getKey());
+      }
+    }
+
+    return result;
+  }
```

Comment:
Should pull this logic out into a service class and write some tests for it.

Reply:
donzo